File: BOJ_2023/1159_농구경기/main.js
// https://www.acmicpc.net/problem/1159

{
  const readline = require('readline')
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  })

  const input = []
  rl.on('line', (line) => {
    input.push(line)
  }).on('close', () => {
    solve(input)
    process.exit()
  })

  function solve(input) {
    const n = Number(input.shift())
    const player = input
    const names = Array(27).fill(0)
    for (let i = 0; i < player.length; i++) {
      names[player[i].charCodeAt(0) - 97]++
    }

    let fir = ''
    for (let i = 0; i < names.length; i++) {
      if (names[i] >= 5) {
        fir += String.fromCharCode(i + 97)
      }
    }
    if (fir === '') console.log('PREDAJA')
    else console.log(fir)
  }
}

File: BOJ_2023/1159_농구경기/main.cpp
// https://www.acmicpc.net/problem/1159

#include <bits/stdc++.h>
using namespace std;
int a[26], n, ret;
string s;
int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  ;
  cin >> n;
  for (int i = 0; i < n; i++) {
    cin >> s;
    a[(int)s[0] - 97]++;
  }

  bool f = false;
  for (int i = 0; i < 26; i++) {
    if (a[i] >= 5) {
      cout << (char)(i + 97);
      f = true;
    }
  }
  if (!f) cout << "PREDAJA";
  return 0;
}